File: Array_II/Q3. Find All Numbers Disappeared in an Array/Q3_FindAllNumbersDisappearedinanArray.py
# Solution class
class Solution(object):

    # Function to find missing numbers
    def findDisappearedNumbers(self, nums):

        # Store length of the list
        n = len(nums)

        # Loop through each index
        for i in range(n):

            # Place numbers in correct position using swapping
            while nums[i] != nums[nums[i] - 1]:

                # Swap current number with its correct position
                nums[nums[i] - 1], nums[i] = nums[i], nums[nums[i] - 1]

        # Collect numbers that are not in correct position
        return [i + 1 for i in range(n) if nums[i] != i + 1]

File: Linear Shoal/Array_II/Q3. Find All Numbers Disappeared in an Array/Q3_FindAllNumbersDisappearedinanArray.java
public class Q3_FindAllNumbersDisappearedinanArray {
    
}
// Solution class
class Solution {

    // Method to find missing numbers in the array
    public List<Integer> findDisappearedNumbers(int[] nums) {

        // Set to store unique numbers from nums
        Set<Integer> set = new HashSet<>();

        // Add each number from nums into the set
        for (int val : nums) {

            // Store value in set
            set.add(val);
        }

        // List to store missing numbers
        ArrayList<Integer> list = new ArrayList<>();

        // Loop from 1 to n (expected numbers)
        for (int i = 1; i <= nums.length; i++) {

            // If number i is not present in set
            if (!set.contains(i)) {

                // Add missing number to list
                list.add(i);
            }
        }

        // Return the list of missing numbers
        return list;
    }
}
